pkgs/top-level/all-packages.nix

```diff
@@ -5633,7 +5633,8 @@ with pkgs;
   wrapRustc = rustc-unwrapped: wrapRustcWith { inherit rustc-unwrapped; };
 
   rust_1_89 = callPackage ../development/compilers/rust/1_89.nix {
-    llvm_20 = llvmPackages_20.libllvm;
+    llvm = llvmPackages_20.libllvm;
+    llvmPackages = llvmPackages_20;
```

Comment:
These should presumably be dropped?

Reply:
That's because rust has been using LLVM 20, does it support 21?

Reply:
I’m confused – if your intent wasn’t to have Rust track the latest LLVM for LLVM 21, then why did you make a commit with the title “rust: track latest LLVM to simplify version bumps” that [made Rust track the latest LLVM](https://github.com/NixOS/nixpkgs/commit/4fda000a019ee8c37da948c85dba706ec4aa4aca), in the LLVM 21 bump PR?

These pins appeared at some point during a rebase. I believe Rust always supports the latest LLVM, but if we want to keep Rust pinned to LLVM 20 then we shouldn’t have a commit talking about unpinning it in an LLVM 21 bump PR that doesn’t unpin it. What makes sense to me is dropping the pins here to accomplish the goal of the commit message, but I might be misunderstanding what the intent of the commit was?

Reply:
Ah, okay. But then the commit nothing to do with LLVM 21 and has no functional effect now, so why is it in the LLVM 21 bump PR?

I thought that the intent of unpinning LLVM in Rust was to update Rust’s LLVM when we update the default LLVM, so pinning it because we’re updating to 21 instead doesn’t make sense to me. If we want to unpin Rust’s LLVM so that it always matches the default LLVM, then we should want Rust to start using LLVM 21 when we make that the default, right?

Reply:
I'm not sure, I'm about to sleep so I can't think. Let's see what rust maintainers think. Maybe @alyssais would have a good idea?

Reply:
Rust should just use our default version of LLVM.